exporter/tanzuobservabilityexporter/transformer_test.go

```diff
@@ -223,6 +224,73 @@ func TestTraceStateTranslatedToTag(t *testing.T) {
 	assert.False(t, ok)
 }
 
+func TestSpanForSourceTag(t *testing.T) {
+	inNanos := int64(50000000)
+
+	//TestCase1: default value for source
+	att := pdata.NewAttributeMap()
+	transform := transformerFromAttributes(att)
+	span := pdata.NewSpan()
+	span.SetSpanID(pdata.NewSpanID([8]byte{0, 0, 0, 0, 0, 0, 0, 1}))
+	span.SetTraceID(pdata.NewTraceID([16]byte{1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1}))
+	span.SetStartTimestamp(pdata.Timestamp(inNanos))
+
+	actual, err := transform.Span(span)
+	require.NoError(t, err, "transforming span to wavefront format")
+	assert.Equal(t, "", actual.Source)
+
+	//TestCase2: source value from resAttrs.source
+	att = pdata.NewAttributeMap()
+	att.InsertString(labelSource, "test_source")
+	att.InsertString(conventions.AttributeHostName, "test_host.name")
+	transform = transformerFromAttributes(att)
+	span = pdata.NewSpan()
+	span.SetSpanID(pdata.NewSpanID([8]byte{0, 0, 0, 0, 0, 0, 0, 1}))
+	span.SetTraceID(pdata.NewTraceID([16]byte{1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1}))
+	span.SetStartTimestamp(pdata.Timestamp(inNanos))
+
+	actual, err = transform.Span(span)
+	require.NoError(t, err, "transforming span to wavefront format")
+	assert.Equal(t, "test_source", actual.Source)
+	assert.Equal(t, "test_host.name", actual.Tags[conventions.AttributeHostName])
+	if value, isFound := actual.Tags[labelSource]; isFound {
+		t.Logf("Tag Source with value " + value + " not expected.")
```

Comment:
Does `t.logF()` cause a test failure? If so, 👍 . If not, I feel the test should fail if the tag hasn't actually been deleted.

Reply:
Added t.Fail(), thank you for pointing this out.